src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala

```diff
@@ -244,7 +244,7 @@ abstract class ExtensionMethods extends Transform with TypingTransformers {
 
           // These three lines are assembling Foo.bar$extension[T1, T2, ...]($this)
           // which leaves the actual argument application for extensionCall.
-          val sel        = Select(gen.mkAttributedRef(companion), extensionMeth)
+          val sel        = Select(gen.mkAttributedRef(companion.owner.thisType, companion), extensionMeth)
```

Comment:
(based on hangout-review:) explain why this prefix is known to be valid (nesting is known by construction)